# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmTable.h
# Commit: 40f861f62a0f
# Full Hash: 40f861f62a0f58040cb6cadf073bc5e97a44231d
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2021-12-23 09:34:36
# Regressor Bug: 1642412
# File Overlap Count: 7
# Description:
#   Bug 1642412 - wasm: Optimize baseline table.size/get/set. r=lth
#   
#   This commit adds a fast path for table.get/set when the table
#   representation is anyref, and an unconditional fast path for
#   table.length.
# ==============================================================================

diff -r 58a45475b524 -r 40f861f62a0f js/src/wasm/WasmTable.h
--- a/js/src/wasm/WasmTable.h	Wed Dec 22 19:09:54 2021 +0200
+++ b/js/src/wasm/WasmTable.h	Wed Dec 22 17:36:29 2021 +0000
@@ -89,8 +89,8 @@
   uint32_t length() const { return length_; }
   Maybe<uint32_t> maximum() const { return maximum_; }
 
-  // Only for function values.  Raw pointer to the table.
-  uint8_t* functionBase() const;
+  // Raw pointer to the table for use in TableTls.
+  uint8_t* tlsElements() const;
 
   // set/get/fillFuncRef is allowed only on table-of-funcref.
   // get/fillAnyRef is allowed only on table-of-anyref.